File: frameworks/gin-rest/internal/models/comment.go
package models

// Comment represents a comment in the system
type Comment struct {
	ID       string `json:"id" db:"id"`
	Content  string `json:"content" db:"content"`
	AuthorID string `json:"author_id" db:"author_id"`
	PostID   string `json:"post_id" db:"post_id"`
	Author   *User  `json:"author,omitempty"`
	Post     *Post  `json:"post,omitempty"`
}

// CommentResponse represents the comment data returned in API responses
type CommentResponse struct {
	ID      string `json:"id"`
	Content string `json:"content"`
	Author  *User  `json:"author,omitempty"`
	Post    *Post  `json:"post,omitempty"`
}

File: frameworks-orm/apollo-orm/src/entities/Comment.ts
import { Entity, PrimaryColumn, Column, ManyToOne } from "typeorm";

@Entity("tb_comments")
export class Comment {
    @PrimaryColumn()
    id: string;

    @Column({ type: "text" })
    content: string;

    @Column({ name: "post_id" })
    postId: string;

    @Column({ name: "author_id" })
    authorId: string;

    // OPTIMIZED: Use eager loading to prevent N+1 queries in GraphQL
    @ManyToOne(() => Post, post => post.comments, {
        eager: true  // Preload post relationship
    })
    post: Post;

    @ManyToOne(() => User, user => user.comments, {
        eager: true  // Preload author relationship
    })
    author: User;
}